tests/utils/utils_test.py

```diff
@@ -572,3 +580,34 @@ def test_kopts_overwrite():
     # Assert
     assert "textmode" in kopts
     assert "info" in kopts
+
+
+@pytest.mark.parametrize(
+    "flock_side_effect,os_open_mock_calls,os_close_mock_calls,expected_exception",
+    [
+        (None, 1, 0, does_not_raise()),
+        (OSError, 2, 1, pytest.raises(OSError)),
+    ],
+)
+def test_filelock(
```

Comment:
This test is very close to the implementation (i.e. will fail as soon as the implementation changes) without actually showing that `utils.filelock()` is a working synchronization tool. IMO it's not worth maintaining in this state.

Reply:
Agree, let me prepare a better test for this.